packages/astro/src/@types/astro.ts

```diff
@@ -1963,12 +1964,13 @@ export interface APIContext<Props extends Record<string, any> = Record<string, a
 	 * ```
 	 */
 	locals: App.Locals;
+	ResponseWithEncoding: typeof ResponseWithEncoding;
 }
 
 export type EndpointOutput =
 	| {
 			body: Body;
-			encoding?: Exclude<BufferEncoding, 'binary'>;
+			encoding?: BufferEncoding;
 	  }
 	| {
 			body: Uint8Array;
```

Comment:
I think this was a mistake in https://github.com/withastro/astro/pull/8003, we technically only support binary strings, and not binary unit8arrays. We have a test for binary strings and the PR caused a type error.

But simple objects are being deprecated anyways so I don't think it's a big deal.